x/stakeibc/keeper/msg_server_restore_interchain_account_test.go

```diff
@@ -105,3 +124,66 @@ func (s *KeeperTestSuite) TestRestoreInterchainAccount_FailsIfAccountExists() {
 	)
 	s.Require().EqualError(err, expectedErrMsg, "registered ica account fails when account already exists")
 }
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_RevertDepositRecords_Failure() {
+	tc := s.SetupRestoreInterchainAccount()
+	s.CreateICAChannel("GAIA.DELEGATION")
+	msg := tc.validMsg
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
+	expectedErrMsg := fmt.Sprintf("existing active channel channel-1 for portID icacontroller-%s.DELEGATION on connection %s for owner %s.DELEGATION: active channel already set for this owner",
+		tc.validMsg.ChainId,
+		s.TransferPath.EndpointB.ConnectionID,
+		tc.validMsg.ChainId,
+	)
+	s.Require().EqualError(err, expectedErrMsg, "registered ica account fails when account already exists")
+	// Verify the deposit record state was reverted
+	for i := 0; i < 2; i++ {
+		depositRecord, found := s.App.RecordsKeeper.GetDepositRecord(s.Ctx(), uint64(i))
+		s.Require().True(found, "deposit record found")
+		s.Require().Equal(recordtypes.DepositRecord_DELEGATION_IN_PROGRESS, depositRecord.Status, "deposit record status should NOT nsg be reverted")
+	}
+}
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_NoRecordChange_Success() {
+	tc := s.SetupRestoreInterchainAccount()
```

Comment:
i'd add a comment at the top of this test that says we're closing and restoring the withdrawal channel and checking that the deposit records do not get reverted